code/modules/mob/living/carbon/human/human.dm

```diff
@@ -150,15 +147,18 @@
 				Paralyse(10)
 
 	take_overall_damage(b_loss,f_loss)
-
+	//attempt to dismember bodyparts
+	for(var/obj/item/bodypart/BP in bodyparts)

```

Comment:
No, a typed for loop runs automatic ```istype()```s on the contents of the list, so filters what you want out of mixed data.
for homogeneous data there's not point ```istype()```ing every element.